LibSWBF2.NET/Wrappers/Level.cs

```diff
@@ -71,22 +75,35 @@ public bool IsWorldLevel
             get { return APIWrapper.Level_IsWorldLevel(NativeInstance); }
         }
 
+        
+        public Terrain[] GetTerrains()
+        {   
+            APIWrapper.Level_GetTerrains(NativeInstance, out IntPtr terrainsArr, out uint numTerrains);
+            Terrain[] terrains = MemUtils.IntPtrToWrapperArray<Terrain>(terrainsArr, (int) numTerrains);
+
+            for (int i = 0; i < numTerrains; i++)
+            {
+                Children.Add(new WeakReference<NativeWrapper>(terrains[i]));
```

Comment:
The `Children.Remove(childRef)` call at line 55 throws an error, when this line is included...
Error text: ```Unhandled Exception:
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.Collections.Generic.HashSet`1+Enumerator[T].MoveNext () [0x00013] in <a63e4875bcd94dc2960d31c0cf2bbd30>:0 
  at LibSWBF2.Wrappers.Level.Delete () [0x00088] in <2a72d9c31e3846478bcc88e5b2ad2f14>:0 
  at LibSWBF2.Wrappers.Level.Finalize () [0x00002] in <2a72d9c31e3846478bcc88e5b2ad2f14>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.Collections.Generic.HashSet`1+Enumerator[T].MoveNext () [0x00013] in <a63e4875bcd94dc2960d31c0cf2bbd30>:0 
  at LibSWBF2.Wrappers.Level.Delete () [0x00088] in <2a72d9c31e3846478bcc88e5b2ad2f14>:0 
  at LibSWBF2.Wrappers.Level.Finalize () [0x00002] in <2a72d9c31e3846478bcc88e5b2ad2f14>:0```

Found some solutions, but none of them are that intuitive so far...

Reply:
Interesting it didn't crash for me...
Just replace with:
```
foreach (WeakReference<NativeWrapper> childRef in Children)
{
    if (childRef.TryGetTarget(out NativeWrapper child))
    {
        child.Invalidate();
    }
}
Children.Clear();
```

Reply:
Lol, I could fix this bug in a heartbeat, but I'll probably waste some time trying to understand it.  Thanks!

Reply:
I guess the real question is, why does ```TryGetTarget``` fail with weakrefs to ```Terrain```?